nixops/deployment.py

```diff
@@ -1272,31 +1323,34 @@ def worker(r):
                     )
 
                     if is_machine(r):
+                        # NOTE: unfortunate mypy doesn't check that
+                        # is_machine calls an isinstance() function
+                        m = cast(nixops.backends.MachineState, r)
```

Comment:
In other places, I replaced is_machine() calls with an instanceof of my own. Maybe ugly ... I wonder if there is a way we could fix this without cast?